array/array_test.go

```diff
@@ -391,7 +391,7 @@ func TestCopyValid(t *testing.T) {
 
 			input := tc.input(mem)
 
-			got := array.CopyValidValues(mem, input).(*array.Int)
+			got := array.CopyValidValues(mem, input, input).(*array.Int)
```

Comment:
It would be good to add some additional tests here for when the null check array and the values array do not have nulls in the same places.